channel/channel.go

```diff
@@ -321,3 +324,22 @@ func (c *Channel) SignAndAddState(s state.State, sk *[]byte) (state.SignedState,
 	}
 	return ss, nil
 }
+
+// UpdateWithChainEvent mutates the receiver if provided with a "new" chain event (with a greater block number than previously seen)
+func (c *Channel) UpdateWithChainEvent(event chainservice.Event) (*Channel, error) {
+	if event.BlockNum() <= c.latestBlockNumber {
+		return c, nil // ignore stale information TODO: is this reorg safe?
+	}
+	c.latestBlockNumber = event.BlockNum()
+	switch e := event.(type) {
+	case chainservice.AllocationUpdatedEvent:
+		c.OnChainFunding[e.AssetAddress] = e.AssetAmount
+	case chainservice.DepositedEvent:
+		c.OnChainFunding[e.Asset] = e.NowHeld
```

Comment:
out of PR scope: `Asset` and `AssetAddress` are a bit jarring when seen this close together!

Reply:
So true :-P